src/lib/sort-modifiers.ts

```diff
@@ -6,33 +6,61 @@ import { AnyConfig } from './types'
  * - When an arbitrary variant appears, it must be preserved which modifiers are before and after it
  */
 export const createSortModifiers = (config: AnyConfig) => {
-    const orderSensitiveModifiers = Object.fromEntries(
-        config.orderSensitiveModifiers.map((modifier) => [modifier, true]),
-    )
+    // Create a lookup Map for O(1) access
+    const sensitiveModifiers = new Map(config.orderSensitiveModifiers.map((mod) => [mod, true]))
 
-    const sortModifiers = (modifiers: string[]) => {
-        if (modifiers.length <= 1) {
-            return modifiers
+    // Binary search insertion point
+    function findInsertionIndex(arr: string[], val: string, start: number, end: number): number {
+        if (start >= end) return start
+
+        const mid = Math.floor((start + end) / 2)
+        if (arr[mid]! < val) {
+            return findInsertionIndex(arr, val, mid + 1, end)
+        } else {
+            return findInsertionIndex(arr, val, start, mid)
         }
+    }
+
+    return function sortModifiers(modifiers: readonly string[]): string[] {
+        // Fast path for common cases
+        if (modifiers.length <= 1) return [...modifiers]
 
-        const sortedModifiers: string[] = []
-        let unsortedModifiers: string[] = []
+        const result: string[] = []
+        let currentSegment: string[] = []
 
-        modifiers.forEach((modifier) => {
-            const isPositionSensitive = modifier[0] === '[' || orderSensitiveModifiers[modifier]
+        // Process modifiers in one pass with binary insertion
+        for (let i = 0; i < modifiers.length; i++) {
+            const modifier = modifiers[i]!
+            const isSensitive = modifier[0] === '[' || sensitiveModifiers.has(modifier)
 
-            if (isPositionSensitive) {
-                sortedModifiers.push(...unsortedModifiers.sort(), modifier)
-                unsortedModifiers = []
+            if (isSensitive) {
+                // Sort and flush current segment
+                if (currentSegment.length > 0) {
+                    result.push(...currentSegment.sort())
+                    currentSegment = []
+                }
+                result.push(modifier)
             } else {
-                unsortedModifiers.push(modifier)
+                // Use binary search insertion for ongoing sorted segment
+                if (currentSegment.length === 0) {
+                    currentSegment.push(modifier)
+                } else {
+                    const pos = findInsertionIndex(
+                        currentSegment,
+                        modifier,
+                        0,
+                        currentSegment.length,
+                    )
+                    currentSegment.splice(pos, 0, modifier)
```

Comment:
I wonder whether this is really faster than sorting once at the end. If we insert elements to anywhere else than the end, the JS engine has to move all subsequent array elements which can be quite inefficient. 🤔